crates/sui-graphql-rpc/src/server/background_tasks.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_graphql::ServerError;
+use tokio::sync::{watch, RwLock};
+use tokio_util::sync::CancellationToken;
+use tracing::{error, info};
+
+use crate::metrics::Metrics;
+use crate::{consistency::Watermark, data::Db};
+
+/// Watermark task that periodically updates the current checkpoint and epoch values.
+pub(crate) struct ServiceWatermarkTask {
+    /// Thread-safe watermark that avoids writer starvation
+    watermark: ServiceWatermark,
+    db: Db,
+    metrics: Metrics,
+    sleep: Duration,
+    cancel: CancellationToken,
+}
+
+pub(crate) type ServiceWatermark = Arc<RwLock<Watermark>>;
+
+/// Starts an infinite loop that periodically updates the `checkpoint_viewed_at` high watermark.
+impl ServiceWatermarkTask {
+    pub(crate) fn new(
+        db: Db,
+        metrics: Metrics,
+        sleep: Duration,
+        cancel: CancellationToken,
+    ) -> Self {
+        let watermark = Arc::new(RwLock::new(Watermark::default()));
+        Self {
+            watermark,
```

Comment:
```suggestion
        Self {
            watermark: Arc::new(RwLock::new(Watermark::default())),
```

...I wonder if you could replace this with the following though:

```suggestion
        Self {
            watermark: Default::default(),
```

Reply:
neat